src/utils/components/FirmwareBootloaderModal/FirmwareBootloaderModal.tsx

```diff
@@ -0,0 +1,112 @@
+import * as React from 'react';
+import produce from 'immer';
+
+import { ensurePath } from '@catalog/utils/WizardVMContext';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { isEqualObject } from '@kubevirt-utils/components/NodeSelectorModal/utils/helpers';
+import TabModal from '@kubevirt-utils/components/TabModal/TabModal';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { Form, FormGroup, Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import {
+  getBIOSBootloader,
+  getBootloaderLabelFromVM,
+  getBootloaderLabels,
+  getUEFIBootloader,
+  isUEFISecureSelected,
+  isUEFISelected,
+} from './utils/utils';
+
+type FirmwareBootloaderModalProps = {
+  vm: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+  onSubmit: (updatedVM: V1VirtualMachine) => Promise<V1VirtualMachine | void>;
+};
+
+const FirmwareBootloaderModal: React.FC<FirmwareBootloaderModalProps> = ({
+  vm,
+  isOpen,
+  onClose,
+  onSubmit,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+
+  const bootloaderLabels = getBootloaderLabels(t);
```

Comment:
const bootloaderOptions = {
bios: {
     title: t('BIOS'),
     description: t('Use BIOS when bootloading the guest OS (Default)')
   },
  uefi: {
      title: t('UEFI'),
      description: t('Use UEFI when bootloading the guest OS (unsecure)'),
      },
  uefiSecure: {
      title: t('UEFI (secure)'),
      description: t('Use secure UEFI boot requires SMM feature, if the SMM feature is not set, choosing this method will set it to true'),
}

Reply:
actually this should be an array, because you loop over it when rendering
```
const bootloaderOptions = [
{
id: 'bios'
title: t('BIOS'),
description: t('Use BIOS when bootloading the guest OS (Default)')
},
{
isd: 'uefi',
title: t('UEFI'),
description: t('Use UEFI when bootloading the guest OS (unsecure)'),
},
{
id: 'uefiSecure',
title: t('UEFI (secure)'),
description: t('Use secure UEFI boot requires SMM feature, if the SMM feature is not set, choosing this method will set it to true'),
}
]
```

Reply:
looking at the loop, `id` should be `value`